react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      imgIndex: 0,
+      activeInterval: null
+    }
+    this.changeIndex = this.changeIndex.bind(this)
+    this.slideShow = this.slideShow.bind(this)
+  }
+  slideShow() {
+    if (this.state.state.activeInterval) {
+      clearInterval(this.state.activeInterval)
+      this.setState({ activeInterval: null })
+    } else {
+      const newInterval = setInterval(this.changeIndex, 1000)
+      this.setState({ activeInterval: newInterval })
+    }
+  }
+  changeIndex(n) {
+    const slides = document.getElementsByClassName("slide");
```

Comment:
You don't want to query the DOM when using React. A better solution here would be to use `this.props.items`, you can check the length of that array to determine the next index.